### PR TITLE
DDCNL-4316 | mongo upgrade

### PR DESCRIPTION
**Pull request by Luka-Abey**
- [x] compiles
- [x] unit tests pass
- [x] integration tests pass
- [x] ATs pass

### CODE REVIEW

File: it/repositories/CachingItSpec.scala [hunk 1]
```diff
@@ -66,35 +61,25 @@ class CachingItSpec extends AnyWordSpecLike with Matchers with GuiceOneAppPerSui
...
       "return an empty list" when {
         "there isn't an existing record" in {
-          val fGet = mongo.get(testNino.withoutSuffix)
+          val fGet = repository.get(testNino.withoutSuffix)
 
           await(fGet) shouldBe List.empty
         }
 
         "there isn't an existing record that matches the requested nino" in {
           val timeOffSet = 10L
```
Comment (Sachin300793) [inline it/repositories/CachingItSpec.scala:78]: val timeOffSet = 10L is unused , we can remove it.

File: it/repositories/CachingItSpec.scala [hunk 2]
```diff
@@ -16,42 +16,37 @@
...
-  with BeforeAndAfterEach {
+class CachingItSpec extends AnyWordSpecLike with Matchers
+  with DefaultPlayMongoRepositorySupport[AddressJourneyTTLModel]
+  with PatienceConfiguration  {
 
-  implicit lazy val ec: ExecutionContext = app.injector.instanceOf[ExecutionContext]
-  implicit val hc = HeaderCarrier(sessionId = Some(SessionId(s"session-${UUID.randomUUID()}")))
 
-  def mongo: EditAddressLockRepository = app.injector.instanceOf[EditAddressLockRepository]
+  lazy val config = mock[ConfigDecorator]
```
Comment (Sachin300793) [inline it/repositories/CachingItSpec.scala:40]: We can remove mocking of ttl if it is unused

File: it/repositories/CachingItSpec.scala (hunk 1 above)
Comment (Sachin300793) [inline it/repositories/CachingItSpec.scala:78]: val timeOffSet = 10L is unused , we can remove it.

File: it/repositories/CachingItSpec.scala (hunk 2 above)
Comment (Sachin300793) [reply it/repositories/CachingItSpec.scala:39]: removed

File: it/repositories/CachingItSpec.scala (hunk 1 above)
Comment (Sachin300793) [reply it/repositories/CachingItSpec.scala]: removed

File: app/repositories/EditAddressLockRepository.scala [hunk 3]
```diff
@@ -17,124 +17,81 @@
...
-  private[repositories] def isTtlSet: Future[Boolean] =
-    for {
-      list <- this.collection.flatMap(_.indexesManager.list())
-    } yield list.exists(_.name == ttlIndex.name)
-
-  val started: Future[Unit] = setIndex().map(_ => ())
+    collection
+      .find(getCore(nino))
+      .toFuture()
+      .map(_.toList)
```
Comment (pascal22p) [inline app/repositories/EditAddressLockRepository.scala:88]: The filter on expire_at is missing

**Review by pascal22p — CHANGES_REQUESTED** [review]

File: it/repositories/CachingItSpec.scala [hunk 4]
```diff
@@ -104,43 +84,28 @@ class CachingItSpec extends AnyWordSpecLike with Matchers with GuiceOneAppPerSui
...
-        }
-      }
-
-      "return the record" when {
-        "there is an existing record and it has not yet expired" in {
-          val currentTime = System.currentTimeMillis()
-          val timeOffSet = 10L
-
-          val nino = testNino.withoutSuffix
+          Thread.sleep(55000)
```
[the remark this answers is not in the document]
Comment (Sachin300793) [reply it/repositories/CachingItSpec.scala]: removed and updated tests

File: app/repositories/EditAddressLockRepository.scala (hunk 3 above)
Comment (Sachin300793) [reply app/repositories/EditAddressLockRepository.scala]: Added filter for expire_at

**Review by pascal22p — APPROVED** [review]

**Pull request merged**